consensus/src/round_manager.rs

```diff
@@ -476,31 +481,83 @@ impl RoundManager {
             .await
     }
 
+    /// Sync to the sync info sending from peer if it has newer commit certificates.
+    /// The result indicates if the sync info has been verified during the execution.
+    async fn sync_up_commit_certs(
+        &mut self,
+        sync_info: &SyncInfo,
+        author: Author,
+    ) -> anyhow::Result<bool> {
+        let local_sync_info = self.block_store.sync_info();
+        if sync_info.has_newer_commit_certificates(&local_sync_info) {
+            info!(
+                self.new_log(LogEvent::ReceiveNewCertificate)
+                    .remote_peer(author),
+                "Local state {}, remote state {}", local_sync_info, sync_info
+            );
+
+            let highest_commit_round = sync_info.highest_commit_round();
+            if self.pending_highest_commit_round.is_none()
+                || highest_commit_round > self.pending_highest_commit_round.unwrap()
+            {
+                // Some information in SyncInfo is ahead of what we have locally.
+                // First verify the SyncInfo (didn't verify it in the yet).
+                sync_info
+                    .verify(&self.epoch_state().verifier)
+                    .map_err(|e| {
+                        error!(
+                            SecurityEvent::InvalidSyncInfoMsg,
+                            sync_info = sync_info,
+                            remote_peer = author,
+                            error = ?e,
+                        );
+                        VerifyError::from(e)
+                    })?;
+                self.pending_highest_commit_round = Some(highest_commit_round);
+                self.block_store
+                    .add_commit_cert(sync_info, self.create_block_retriever(author))
+                    .await;
+                return Ok(true);
+            }
+        }
+        Ok(false)
+    }
+
     /// Sync to the sync info sending from peer if it has newer certificates.
-    async fn sync_up(&mut self, sync_info: &SyncInfo, author: Author) -> anyhow::Result<()> {
+    async fn sync_up_non_commit_certs(
+        &mut self,
+        sync_info: &SyncInfo,
+        author: Author,
+        sync_info_verified: bool,
+    ) -> anyhow::Result<()> {
         let local_sync_info = self.block_store.sync_info();
-        if sync_info.has_newer_certificates(&local_sync_info) {
+        if sync_info.has_newer_non_commit_certificates(&local_sync_info) {
+            // TODO: Duplicate info message. Check if it is necessary.
```

Comment:
we should have separate log entries for this

Reply:
Done.